### PR TITLE
Various small  edits in ImageFromSetAdd and IO ITK  

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description
Improved ITK image selection in ImageSelector and add ITK xx.gz support.
 New option to keep set domain in ImageFromSet. 

# Checklist

- [ ] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [ ] Doxygen documentation of the code completed (classes, methods, types, members...)
- [ ] Documentation module page added or updated.
- [ ] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [ ] No warning raised in Debug mode.
- [ ] All continuous integration tests pass (Github Actions & appveyor)

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -34,7 +34,12 @@
     polygonal surfaces (digital surfaces, or any PolygonalMesh instance) (David
     Coeurjolly, [#1603]((https://github.com/DGtal-team/DGtal/pull/1603))
 
+  
 ## Changes
+- *Image*
+  - Improved ITK image selection in ImageSelector and add ITK xx.gz support.
+    New option to keep set domain in ImageFromSet.
```
Comment (JacquesOlivierLachaud) [inline ChangeLog.md:41]: "New option to keep set domain or to compute current bounding box of elements of the set in ImageFromSet"

**Review by JacquesOlivierLachaud — APPROVED** [review]
Looks fine to me. Not easy to write a catch test without ITK files.
Have you checked on your computer ?

**Comment by kerautret** [comment]
Thanks @JacquesOlivierLachaud 
> Looks fine to me. Not easy to write a catch test without ITK files.
> Have you checked on your computer ?

Yes, and I just change a little the extension file checks in order to avoid to edit various list when a new format is added. I added the list extension in ITKIOTrait.h perhaps not the best place but perhaps better than common.h? By the way last commits includes also tests for all IKT formats at least for writeré. And sure It was tested on my Mac on linux with @JonasLamy

**Comment by kerautret** [comment]
Thanks @JacquesOlivierLachaud can I merge with last new commit ?

**Comment by dcoeurjo** [comment]
Hi @kerautret,i don't see the ITKIOTraits in your PR ? Did I misunderstand something ?

**Review by dcoeurjo — APPROVED** [review]

**Comment by kerautret** [comment]
> Hi @kerautret,i don't see the ITKIOTraits in your PR ? Did I misunderstand something ?

Thanks sorry I forget to push ;)

**Comment by kerautret** [comment]
@dcoeurjo That should be fine for merging I think, I fix the codify and add the last commit  ITKIOTrait

**Review by dcoeurjo — APPROVED** [review]
All good to me

**Comment by kerautret** [comment]
Thanks ! merging :)

**Pull request merged**